projects/packages/forms/src/modules/file-field/view.js

```diff
@@ -0,0 +1,154 @@
+/**
+ * WordPress dependencies
+ */
+import { store, getContext } from '@wordpress/interactivity';
+const addFileToContext = (context, file) => {
+	const reader = new FileReader();
+	reader.readAsDataURL(file);
+	reader.onload = () => {
+		file.url = 'url(' + reader.result + ')';
+		file.id = performance.now() + '-' + Math.random();
+		file.formattedSize = formatBytes(file.size, 2, state.i18n.locale);
+		file.hasToken = false;
+		context.files.push(file);
+		context.hasFiles = true;
+		FileHandler.uploadFile(file, context);
+	};
+};
+
+const formatBytes = (size, decimals = 2, locale = 'en-US') => {
+	if (size === 0) return '0 bytes';
+	const k = 1024;
+	const dm = decimals < 0 ? 0 : decimals;
+	const sizes = state.i18n.fileSizeUnits || ['Bytes', 'KB', 'MB', 'GB', 'TB'];
+	const i = Math.floor(Math.log(size) / Math.log(k));
+	const formattedSize = parseFloat((size / Math.pow(k, i)).toFixed(dm));
+	const numberFormat = new Intl.NumberFormat(locale, {
+		minimumFractionDigits: dm,
+		maximumFractionDigits: dm,
+	});
+	return `${numberFormat.format(formattedSize)} ${sizes[i]}`;
+};
+const { state } = store('jpDropZone', {
+	state: {},
+	actions: {
+		openFilePicker: event => {
+			const fileInput = event.target.parentNode.querySelector('.jetpack-form-file-field');
+			if (fileInput) {
+				fileInput.click();
+			}
+		},
+
+		fileAdded: event => {
+			const context = getContext();
+			const files = Array.from(event.target.files);
+			files.forEach(addFileToContext.bind(null, context));
+		},
+
+		fileDropped: event => {
+			const context = getContext();
+			if (event.dataTransfer) {
+				for (const item of Array.from(event.dataTransfer.items)) {
+					if (item.webkitGetAsEntry()?.isDirectory) {
+						return;
+					}
+					addFileToContext(context, item.getAsFile());
+				}
+			}
+			context.isDropping = false;
+			event.preventDefault();
+		},
+		dragOver: event => {
+			const context = getContext();
+			context.isDropping = true;
+			event.preventDefault();
+		},
+		dragLeave: () => {
+			const context = getContext();
+			context.isDropping = false;
+		},
+		removeFile: event => {
+			const context = getContext();
+			const fileId = event.target.dataset.id;
+			context.files = context.files.filter(file => file.id !== fileId);
+			context.hasFiles = context.files.length > 0;
+		},
+	},
+	callbacks: {
+		logCounter: () => {
+			const { files } = getContext();
+			console.log('howdy!!!!', files);
+		},
+	},
+});
+
+const updateFile = file => {
+	const context = getContext();
+	const index = context.files.findIndex(f => f.id === file.id);
+	context.files[index] = file;
+	console.log('howdy!updateFile !!!', { index, file, context });
```

Comment:
I know it's WIP but just flagging for removal so not to forget it.

Reply:
Looks like these have been removed.